app/controllers/individual/tasklist/addbusiness/YourIncomeSourceToSignUpController.scala

```diff
@@ -41,11 +41,14 @@ class YourIncomeSourceToSignUpController @Inject()(yourIncomeSourceToSignUp: You
 
   def show: Action[AnyContent] = Authenticated.async { implicit request =>
     _ =>
-      referenceRetrieval.getIndividualReference flatMap { reference =>
-        subscriptionDetailsService.fetchAllIncomeSources(reference) map { incomeSources =>
-          Ok(view(
-            incomeSources
-          ))
+      for {
+        reference <- referenceRetrieval.getIndividualReference
+        incomeSources <- subscriptionDetailsService.fetchAllIncomeSources(reference)
+        maybePrePop <- subscriptionDetailsService.fetchPrePopFlag(reference)
+      } yield {
+        maybePrePop match {
+          case Some(prePopFlag) => Ok(view(incomeSources, prePopFlag))
+          case None => Ok(view(incomeSources, isPrePopulated = false))
         }
```

Comment:
```suggestion
        Ok(view(incomeSources, maybePrePop.contains(true)))
```
Can be simplified like so.

`maybePrePop.contains(true)` will return:
- true when defined and true
- false when defined and false
- false when not defined